src/components/QuestionsDisplay.tsx

```diff
@@ -1,59 +1,90 @@
+    props.integrations,
+    props.tags,
+    props.search,
+    (props.filter === 'any') ? FilterType.ANY : FilterType.ALL,
+    props.categories
+  ), [props])
 
   return (
-    <Paper className={classes.root} style={{margin: props.center ? 'auto' : ''}}>
+    <Card elevation={0} className={windowSize.width > 750 ? classes.root : classes.smallRoot} style={{margin: props.center ? 'auto' : ''}}>
+      <Box style={{textAlign: 'right'}} mr={1} mb={-3}>
+        <em>{filteredQuestions.length} of {props.managedQuestions.questions.length}</em>
+      </Box>
       {filteredQuestions.length !== 0 ?
         (
           <div>
-            {filteredQuestions.map((question: Question, index: number) => {
-              return (
-                <Link key={index} style={{textDecoration: 'none'}} to={`/question/${hash.sha1().update(question.title).digest('hex')}`}>
-                  <Paper elevation={0} style={{display: 'flex', marginBottom: '0.5%', padding: '0.5%'}}>
-                    <span className={classes.item}>{question.title}</span>
-                    <Icon className={classes.arrow}><ArrowForwardIosIcon/></Icon>
-                    <Divider/>
-                  </Paper>
-                </Link>
-              )})
-            }
-            {(props.questionNumber < props.managedQuestions.questions.length && !(props.integration !== 'none' || props.tags.length !== 0 || props.search !== '')) ? (
-              <Button color='primary' variant='contained' onClick={() => props.setQuestionNumber((num: number) => num + 10)}>Load more</Button>
-            ) : null}
+            {[...props.allCategories, undefined].map(category =>
+              <div key={category}>
+                {filteredQuestions.filter(question =>
+                  question.category === category
+                ).length !== 0 ?
+                  <Box m={1} mt={2}>
+                    <Typography variant='h6'>{category === undefined ? 'No Category' : category}</Typography>
+                  </Box>
+                : (
+                  null
+                )}
+
+                {filteredQuestions.filter(question =>
+                  question.category === category
+                ).map((question: Question, index: number) =>
+                  <>
+                    <Box key={index} onClick={() => history.push(`/question/${question.hash}`)} style={{display: 'flex'}}>
+                      <span className={classes.item}>
+                        {question.title}
+                        <div>
+                          {/* question.tags ? question.tags.map(tag => <Chip className={classes.chip} label={tag}/>) : null */}
+                        </div>
+                      </span>
+                      <Icon className={classes.arrow}><ArrowForwardIosIcon/></Icon>
+                    </Box>
+                    {index !== filteredQuestions.filter(question =>
+                      question.category === category
+                    ).length - 1 ? <Divider/> : <span/>}
```

Comment:
You want to avoid having inline logic here. Use a function with a descriptive name at least.